pip_audit/_dependency_source/pip.py

```diff
@@ -87,6 +90,27 @@ def collect(self) -> Iterator[Dependency]:
         except Exception as e:
             raise PipSourceError("failed to list installed distributions") from e
 
+    def fix(self, fix_version: ResolvedFixVersion) -> None:
+        """
+        Fixes a dependency version in this `PipSource`.
+        """
+        fix_cmd = [
+            sys.executable,
+            "-m",
+            "pip",
+            "install",
+            f"{fix_version.dep.canonical_name}=={fix_version.version}",
+        ]
+        try:
+            subprocess.run(
+                fix_cmd, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+            )
+        except subprocess.CalledProcessError as cpe:
+            raise PipSourceError(
```

Comment:
Nitpick: we might want to consider creating a specialized `DependencyFixError` that each dependency source in turn specializes.

So the hierarchy would be: `DependencySourceError -> DependencyFixError -> PipFixError | RequirementFixError`.